presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitPartitionInfo.java

```diff
@@ -45,16 +50,44 @@
 
     public HiveSplitPartitionInfo(
             Properties schema,
+            URI path,
             List<HivePartitionKey> partitionKeys,
             String partitionName,
             Map<Integer, HiveTypeName> columnCoercions,
             Optional<HiveSplit.BucketConversion> bucketConversion)
     {
-        this.schema = requireNonNull(schema, "schema is null");
-        this.partitionKeys = requireNonNull(partitionKeys, "partitionKeys is null");
-        this.partitionName = requireNonNull(partitionName, "partitionName is null");
-        this.columnCoercions = requireNonNull(columnCoercions, "columnCoersions is null");
-        this.bucketConversion = requireNonNull(bucketConversion, "bucketConversion is null");
+        requireNonNull(schema, "schema is null");
```

Comment:
Why to split `requireNonNull`s from the assignments?

Reply:
I don't feel strongly, but I think it's harder to read when you have requireNonNull embedded inside functions.

Reply:
We have many places when we do something like `ImmutableList.copyOf(requireNonNull(someList, "someList is null"))`

When `requireNonNull` and assignment are on different lines it is easier to forget add a check, or it is easier to have them out of order. 

But anyhow, i also don't feel strongly. Feel free to leave it as is.